src/Core/Composer/ModulesConfigurationProcessor.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * 2007-2019 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Composer;
+
+use Composer\IO\IOInterface;
+use Symfony\Component\Process\Process;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+
+final class ModulesConfigurationProcessor
```

Comment:
I'm not very sure about the responsibility of this class, is it supposed to interpret the configuration in the composer.json file to extract `ModuleInformation`? is it supposed to launch composer processes?

I know you mentioned it is a WIP and that composer processes are gonna be extracted from here.. but it's still a bit blur what is done by what..

I think you could have some kind of `ModuleInformationListFactory` (quick naming) which would return the list of modules from the config file, or config array (`ModuleInformation` could contain the module path)

And a `ModulesInstaller` class which would launch the processes for all the listed modules (you could have a `ModulesInstallerInterface` implemented by `ComposerModulesInstaller`, `AddonsModulesInstaller`, ...)

Reply:
> I think you could have some kind of ModuleInformationListFactory (quick naming) which would return the list of modules from the config file, or config array (ModuleInformation could contain the module path)

We retrieve the information from Composer and I don't want to implement my own way to do it.

* Composer stuff: their tests, the code they maintain => will work forever
* Our own stuff: we need to test and maintain it => hmm...

> And a ModulesInstaller class which would launch the processes for all the listed modules (you could have a ModulesInstallerInterface implemented by ComposerModulesInstaller, AddonsModulesInstaller, ...)

No, but for another reason: the scope of this feature is really clear: be able - in dev environment - to install modules using Composer "install" and "update" commands.
I won't open a contract that allows providing the same features outside of Composer in this package.
This is a Composer extension, not less and not more :+1:

Reply:
alright!